consul/pool.go

```diff
@@ -265,7 +269,8 @@ type HalfCloser interface {
 // given connection timeout.
 func (p *ConnPool) DialTimeout(dc string, addr net.Addr, timeout time.Duration) (net.Conn, HalfCloser, error) {
 	// Try to dial the conn
-	conn, err := net.DialTimeout("tcp", addr.String(), defaultDialTimeout)
+	d := &net.Dialer{LocalAddr: p.src, Timeout: timeout}
```

Comment:
Pls note that this code uses the `timeout` value passed into the function instead of `defaultDialTimeout` as before.